src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,215 @@
+package wooteco.subway.domain;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class Sections {
+
+    private final List<Section> values;
+
+    public Sections(Section section) {
+        values = new LinkedList<>();
+        values.add(section);
+    }
+
+    public Sections(List<Section> sections) {
+        values = new LinkedList<>(sortedSections(sections));
+    }
+
+    private List<Section> sortedSections(List<Section> sections) {
+        if (sections.size() <= 1) {
+            return sections;
+        }
+        Long upDestinationId = findUpDestinationId(sections);
+        Map<Long, Section> sectionsByIds = sections.stream()
+            .collect(Collectors.toMap(Section::getUpStationId, section -> section, (a, b) -> b));
+
+        return getSortedValuesBy(upDestinationId, (key) -> sectionsByIds.get(key).getDownStationId(), sectionsByIds);
+    }
+
+    private Long findUpDestinationId(List<Section> sections) {
+        List<Long> upStations = collectValuesBy(Section::getUpStationId, sections);
+        List<Long> downStations = collectValuesBy(Section::getDownStationId, sections);
+
+        upStations.removeAll(downStations);
+        if (upStations.size() != 1) {
+            throw new IllegalStateException("저장된 구간들 값이 올바르지 않습니다.");
+        }
+        return upStations.get(0);
+    }
```

Comment:
방식이 굉장히 독특하네요 ㅎㅎ
그런데, "저장된 구간들 값이 올바르지 않습니다."라는 메시지만으로 상황을 잘 파악할 수 있을까요?

Reply:
상행 종점을 찾는 메서드인데 오류 메시지는 관련 내용이 없어서 파악하기 힘들군요🤔. 메서드가 하는 일을 담도록 변경해보았습니다. 감사합니다!!

```suggestion
    private Long findUpDestinationId(List<Section> sections) {
        List<Long> upStations = collectValuesBy(Section::getUpStationId, sections);
        List<Long> downStations = collectValuesBy(Section::getDownStationId, sections);

        upStations.removeAll(downStations);
        if (upStations.size() != 1) {
            throw new IllegalStateException("상행 종점을 찾을 수 없습니다. 구간 목록이 올바르게 저장되지 않습니다.");
        }
        return upStations.get(0);
    }
```